app/src/__utils__/safe-integer.ts

```diff
@@ -0,0 +1,78 @@
+type numeric = number | bigint;
+
+export class SafeInteger {
+	private _value: numeric;
+	private MAX_VALUE: numeric;
+	private MIN_VALUE: numeric;
+
+	constructor(value: number | bigint | string, isBigInt = false) {
+		if (isBigInt) {
+			this._value = BigInt(value);
+			this.MAX_VALUE = BigInt(9223372036854775807n);
+			this.MIN_VALUE = BigInt(-9223372036854775807n);
```

Comment:
Let's define these as constants outside of the class so these big numbers only get initialized once not for every use of the class.

And dont forget to add some comments about what these limits represent considering they are not bigInt limits as such.